WatchdogService/CleanupOperations.cs

```diff
@@ -230,22 +229,24 @@ internal async Task CleanupDiagnosticTablesAsync()
             {
                 // Create the storage credentials and connect to the storage account.
                 // The SAS token must be a Table service SAS URL with permissions to read, delete and list table entries. HTTPS must be used.
-                StorageCredentials sc = new StorageCredentials(this._sasToken);
-                CloudTableClient client = new CloudTableClient(new StorageUri(new Uri(this._endpoint)), sc);
+                AzureSasCredential sc = new AzureSasCredential(this._sasToken);
+                TableServiceClient client = new TableServiceClient(new Uri(this._endpoint), sc);
 
                 // Inspect each table for items to be removed.
                 foreach (string tableName in this._tablesToInspect)
                 {
-                    CloudTable table = client.GetTableReference(tableName);
-                    if (true == await table.ExistsAsync(this._token))
-                    {
-                        await this.EnumerateTableItemsAsync(table);
+                    await foreach (var table in client.QueryAsync()) {
+                        if (table.Name == tableName)
+                        {
+                            var tableClient = client.GetTableClient(table.Name);
+                            await this.EnumerateTableItemsAsync(table, client);
```

Comment:
We could pass the tableClient to EnumerateTableItemsAsync
```suggestion
                            var tableClient = client.GetTableClient(table.Name);
                            await this.EnumerateTableItemsAsync(table, tableClient);
```